Copyright 2025 The zb Authors
SPDX-License-Identifier: GPL-2.0-only

Use HOSTLDFLAGS for executables built from a single .c file.
--- busybox-1.36.1.orig/scripts/Makefile.host
+++ busybox-1.36.1/scripts/Makefile.host
@@ -98,7 +98,7 @@ hostcxx_flags  = -Wp,-MD,$(depfile) $(__
 # Create executable from a single .c file
 # host-csingle -> Executable
 quiet_cmd_host-csingle 	= HOSTCC  $@
-      cmd_host-csingle	= $(HOSTCC) $(hostc_flags) -o $@ $< \
+      cmd_host-csingle	= $(HOSTCC) $(HOSTLDFLAGS) $(hostc_flags) -o $@ $< \
 			  $(HOST_LOADLIBES) $(HOSTLOADLIBES_$(@F))
 $(host-csingle): %: %.c FORCE
 	$(call if_changed_dep,host-csingle)
